pilot/pkg/bootstrap/config_compare.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap
+
+import (
+	"reflect"
+
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/schema/gvk"
+)
+
+// EqualsFunc defines a comparison func for an API proto.
+type EqualsFunc func(prev config.Config, curr config.Config) bool
+
+var equalsFuncs = make(map[string]EqualsFunc)
+
+func registerEqualsFunc(name string, equals EqualsFunc) {
+	equalsFuncs[name] = equals
+}
+
+// destinationRuleEquals checks proxy policies
+var destinationRuleEquals = func(prev config.Config, curr config.Config) bool {
+	prevspec := prev.Spec
+	currspec := curr.Spec
+	// TODO(ramaraochavali): optimize this by comparing relevant fields only.
```

Comment:
why not in this PR?

Reply:
and maybe we should have a generic specEquals if that is all the functionality in virtualServiceEquals and destinationRuleEquals, and register the specEquals function under the DR and VS GVK

Reply:
My main concern is using `reflect.DeepEquals` for performance reasons. So I think we need an optimized version of specEquals that compares each field.

That is what I meant by relevant fields. 

@howardjohn  yes. I also think all fields are relevant - but I want to optimize the field comparisions though. That is why I left that TODO.